### PR TITLE
Feature/invoice detail page

### PR DESCRIPTION
**Pull request by mugnivenko**
https://app.clickup.com/t/1t1wu83
layout + integration

### CODE REVIEW

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]

File: api/SWR/payRuns.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import { useFetchWithParams } from './useFetchWithParams';
 import useGetData from './useGetData';
+import { getUrlOrNull } from '../Utils/FuncUtils';
 
 const PAY_RUNS_URL = '/payruns';
-const getPayRunUrl = (payRunId) => `${PAY_RUNS_URL}/${payRunId}`;
+const getPayRunUrl = (payRunId, additionalString = '') => getUrlOrNull(`${PAY_RUNS_URL}/${payRunId}${additionalString}`);
```
[the remark this answers is not in the document]
Comment (mugnivenko) [reply api/SWR/payRuns.js:6]: when it includes undefined string

[the remark this answers is not in the document]
Comment (mugnivenko) [reply api/SWR/payRuns.js:6]: we always write id !== undefined ? url : null. So i decided to create a function for that case

File: api/SWR/payRuns.js [hunk 2]
```diff
@@ -18,14 +19,22 @@ export const useInvoiceListView = ({ payRunId, params }) =>
     errorText: 'Cannot get invoice list',
   });
 
-export const getSinglePayrun = ({ payRunId }) => useGetData(payRunId !== undefined ? getPayRunUrl(payRunId) : null);
+export const usePayRunInvoice = (payRunId, invoiceId) =>
+  useGetData(getPayRunUrl(payRunId, `/invoices/${invoiceId}`))
```
[the remark this answers is not in the document]
Comment (mugnivenko) [reply api/SWR/payRuns.js:23]: if so, we 100% need to change all the url that depends on payrun

File: pages/payruns/[guid]/invoice/[id].js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  const router = useRouter();
+  const { guid: payRunId, id: invoiceId } = router.query;
+  const { data: invoice, isLoading } = usePayRunInvoice(payRunId, invoiceId);
+
+  return (
+    <Container className="invoice-detail">
+      <DynamicBreadcrumbs />
+      <Loading isLoading={isLoading} />
+      <Container className="default-container invoice-detail__main-container">
+        <h3 className="invoice-detail--title">Pay run period</h3>
```
Comment (akeamwoodsND) [inline pages/payruns/[guid]/invoice/[id].js:35]: use Heading component

File: pages/payruns/[guid]/invoice/[id].js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        <p
+          className="invoice-detail--sub-title">{getEnGBFormattedDate(dateFrom)}{dateTo ? ` - ${getEnGBFormattedDate(dateTo)}` : ''}</p>
```
Comment (akeamwoodsND) [inline pages/payruns/[guid]/invoice/[id].js:37]: use Hint component

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]

File: styles/css/Finance/invoice-detail.scss [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    text-align: center;
+  }
+
+  .link-button.green {
+    font-size: 19px;
+    font-weight: 400;
+  }
+}
+
+.invoice-detail--title {
```
Comment (akeamwoodsND) [inline styles/css/Finance/invoice-detail.scss:13]: delete

File: pages/payruns/[guid]/invoice/[id].js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  const { data: invoice, isLoading } = usePayRunInvoice(payRunId, invoiceId);
+
+  return (
+    <Container className="invoice-detail">
+      <DynamicBreadcrumbs />
+      <Loading isLoading={isLoading} />
+      <Container className="default-container invoice-detail__main-container">
+        <Heading fontWeight={400} size='xl'>Pay run period</Heading>
+        <HorizontalSeparator height={9} />
+        <HintDate dateFrom={invoice.startDate} dateTo={invoice.endDate} />
```
Comment (akeamwoodsND) [inline pages/payruns/[guid]/invoice/[id].js:34]: delete this component, use normal Hint component

File: pages/payruns/[guid]/invoice/[id].js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  const router = useRouter();
+  const { guid: payRunId, id: invoiceId } = router.query;
+  const { data: invoice, isLoading } = usePayRunInvoice(payRunId, invoiceId);
+
+  return (
+    <Container className="invoice-detail">
+      <DynamicBreadcrumbs />
+      <Loading isLoading={isLoading} />
+      <Container className="default-container invoice-detail__main-container">
+        <Heading fontWeight={400} size='xl'>Pay run period</Heading>
```
Comment (akeamwoodsND) [inline pages/payruns/[guid]/invoice/[id].js:32]: remove font weight

Comment (akeamwoodsND) [reply pages/payruns/[guid]/invoice/[id].js:32]: we want the headings on pages to look the same

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
breadcrumbs link is wrong, there's a console warning about it
Screenshot 2021-11-17 at 10 04 35

**Comment by mugnivenko** [comment]
> breadcrumbs link is wrong, there's a console warning about it Screenshot 2021-11-17 at 10 04 35

it is because DynamicBreadcrumbs use localStorage. I fixed it, by rendering DynamicBreadcrumbs only on frontend

**Pull request merged**